zpm.c

```diff
@@ -227,13 +249,32 @@ char* get_zpm_plugin_list() {
         strcpy(listing, "Nothing to show.");
     }
 
+    free(plugin_list_path);
+
     return listing;
 }
 
 int plugins_update_local_clone() {
-    // iterate over plugins_list
-    // exec git pull --git-dir=
-    return 0;
+    int ret;
+    char* command = malloc(PATH_MAX);
+    char* listing = get_zpm_plugin_list();
+    char* plugin_name = strtok(listing, "\n");
+
+    if(!strcmp(listing, "Nothing to show.")) {
+      printf("Nothing to update.");
+      return -1;
+    }
+    printf("Updating plugins ...\n");
+    while(plugin_name) {
+      strcpy(command, "cd ~/.zpm/plugins/");
+      strcat(command, plugin_name);
+      strcat(command, "; git pull");
```

Comment:
Adding a print below this line would be the final touch up:

    printf("Updating %s...\n", plugin_name);